pkg/controller/elasticsearch/driver/upgrade_pods_deletion.go

```diff
@@ -78,7 +78,39 @@ func (ctx *rollingUpgradeCtx) Delete() ([]corev1.Pod, error) {
 			return deletedPods, err
 		}
 
-		if err := deletePod(ctx.client, ctx.ES, podToDelete, ctx.expectations); err != nil {
+		if err := deletePod(ctx.client, ctx.ES, podToDelete, ctx.expectations, "Deleting pod for rolling upgrade"); err != nil {
+			return deletedPods, err
+		}
+		deletedPods = append(deletedPods, podToDelete)
+	}
+	return deletedPods, nil
+}
+
+// DeleteAll unconditionally deletes all upgradeable Pods after calling the node shutdown API and accounting for quorum
+// changes on older versions of Elasticsearch as applicable.
+func (ctx *upgradeCtx) DeleteAll() ([]corev1.Pod, error) {
+	if len(ctx.podsToUpgrade) == 0 {
+		return nil, nil
+	}
+
+	if err := ctx.prepareClusterForNodeRestart(ctx.podsToUpgrade); err != nil {
+		return nil, err
+	}
+
+	for _, podToDelete := range ctx.podsToUpgrade {
+		if err := ctx.handleMasterScaleChange(podToDelete); err != nil {
+			return nil, err
+		}
+		// do not delete any Pods if at least one is not ready for deletion
+		if readyToDelete, err := ctx.readyToDelete(podToDelete); err != nil || !readyToDelete {
```

Comment:
Should we surface the situation where `!readyToDelete` is true, with a log message for example ? I'll try to see how to deal with it in the context of #5328